libraries/resource_firewall_rule.rb

```diff
@@ -22,6 +22,9 @@ class Resource::FirewallRule < Resource
     attribute(:redirect_port, :kind_of => Integer)
     attribute(:description, :kind_of => String, :name_attribute => true)
 
+    # Indicate whether this rule needs to be persistent on the system
+    attribute(:persist, :kind_of => Integer, :equal_to => [0, 1], :default => 0)
```

Comment:
Yes, more work needs to be done to apply this feature to ufw and iptables. There is another cookbook https://github.com/opscode-cookbooks/iptables to persist individual rules within template files under `/etc/iptables.d/` directory, and then write these rules to `/etc/sysconfig/iptables`. Maybe we can use similar idea to write individual rules to `/etc/sysconfig/iptables`.

As for ufw, maybe we need to apply these rules to `/etc/ufw/before.rules` or `/etc/ufw/after.rules`.